src/vda5050_connector/vda5050_connector.cpp

```diff
@@ -50,11 +50,18 @@ VDA5050Connector::VDA5050Connector() : state(State()), order(Order()) {
     ROS_ERROR("%s not found in the configuration!", SN_PARAM);
   }
 
-  stateTimer = nh.createTimer(ros::Duration(0.8), std::bind(&VDA5050Connector::PublishState, this));
-  visTimer =
-      nh.createTimer(ros::Duration(0.3), std::bind(&VDA5050Connector::PublishVisualization, this));
+  ros::NodeHandle private_nh("~");
+  double stateMsgPeriod, visMsgPeriod, connMsgPeriod;
+  private_nh.param<double>("publish_periods/state_msg", stateMsgPeriod, 0.8);
+  private_nh.param<double>("publish_periods/visualization_msg", visMsgPeriod, 0.3);
+  private_nh.param<double>("publish_periods/conn_msg", connMsgPeriod, 15.0);
```

Comment:
No need to create a NodeHandle here. The class already has a NodeHandle attribute that you can use to get the params you need from the config file. 

Check LinkPublishTopics() for an example

Reply:
Using a private NodeHandle is quite common to resolve names in the private namespace of the Node, especially when working with parameters.